core/src/test/java/org/apache/iceberg/metrics/TestMetricsReporters.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.junit.jupiter.api.Test;
+
+public class TestMetricsReporters {
+  @Test
+  public void combineWithNullReporter() {
+    MetricsReporter reporter = report -> {};
+    assertThat(MetricsReporters.combine(null, null)).isNull();
+    assertThat(MetricsReporters.combine(null, reporter)).isSameAs(reporter);
+    assertThat(MetricsReporters.combine(reporter, null)).isSameAs(reporter);
+  }
+
+  @Test
+  public void combineSame() {
+    MetricsReporter reporter = LoggingMetricsReporter.instance();
+    assertThat(MetricsReporters.combine(reporter, reporter)).isSameAs(reporter);
+  }
+
+  @Test
+  public void combineSameButDifferentInstances() {
```

Comment:
Doesn't "same" usually mean the same instance, as in the `isSameAs` checks above? I think this is referring to the same class, but different instances.

Reply:
indeed the naming is slightly confusing here. We want to test reporters of the same class but different instances. I've updated the naming